exchanges/okgroup/okgroup_websocket.go

```diff
@@ -494,36 +520,38 @@ func (o *OKGroup) wsProcessCandles(response *WebsocketDataResponse) {
 		}
 		klineData.OpenPrice, err = strconv.ParseFloat(response.Data[i].Candle[1], 64)
 		if err != nil {
-			o.Websocket.DataHandler <- err
-			continue
+			return err
 		}
 		klineData.HighPrice, err = strconv.ParseFloat(response.Data[i].Candle[2], 64)
 		if err != nil {
-			o.Websocket.DataHandler <- err
-			continue
+			return err
 		}
 		klineData.LowPrice, err = strconv.ParseFloat(response.Data[i].Candle[3], 64)
 		if err != nil {
-			o.Websocket.DataHandler <- err
-			continue
+			return err
 		}
 		klineData.ClosePrice, err = strconv.ParseFloat(response.Data[i].Candle[4], 64)
 		if err != nil {
-			o.Websocket.DataHandler <- err
-			continue
+			return err
 		}
 		klineData.Volume, err = strconv.ParseFloat(response.Data[i].Candle[5], 64)
 		if err != nil {
-			o.Websocket.DataHandler <- err
-			continue
+			return err
 		}
 
 		o.Websocket.DataHandler <- klineData
 	}
+	return nil
 }
 
 // WsProcessOrderBook Validates the checksum and updates internal orderbook values
-func (o *OKGroup) WsProcessOrderBook(response *WebsocketDataResponse) (err error) {
+func (o *OKGroup) WsProcessOrderBook(respRaw []byte) error {
+	var response WebsocketOrderBooksData
+	err := json.Unmarshal(respRaw, &response)
+	if err != nil {
+		return err
+	}
+	orderbookMutex.Lock()
```

Comment:
should use defer orderbookMutex.Unlock() here, many code paths here which could potentially lock this for eternity